schemas/v1.0/azure.yaml.json

```diff
@@ -132,21 +132,48 @@
                         }
                     }
                 },
-                "if": {
-                    "not": {
+                "required": ["project"],
+                "allOf": [
+                    {
+                        "if": {
+                            "not": {
+                                "properties": {
+                                    "host": {
+                                        "const": "containerapp"
+                                    }
+                                }
+                            }
+                        },
+                        "then": {
+                            "properties": {
+                                "docker": false
+                            }
+                        }
+                    },
+                    {
+                        "if": {
+                            "properties": {
+                                "language": { "const": "java" }
+                            }
+                        },
+                        "then": {
+                            "properties": {
+                                "dist": {
+                                    "type": "string",
+                                    "description": "The CLI will use the JAR file in this directory to create the deployment artifact (ZIP file). If omitted, the CLI will detect the output directory based on the build system in-use."
+                                }
+                            }
+                        }
+                    },
+                    {
                         "properties": {
-                            "host": {
-                                "const": "containerapp"
+                            "dist": {
+                                "type": "string",
+                                "description": "The CLI will use files under this path to create the deployment artifact (ZIP file). If omitted, all files under service project directory will be included."
                             }
```

Comment:
Is this whole change primarily to support a different description in the schema doc?  If so we may just want to have a more generic message about the `dist` property that can apply to any language?

Reply:
We can, but then we can't get the exact semantics correct -- we would only be able to describe abstractly how it works. I think the extra detail here is important for the user to configure the setting correctly.